staging/src/k8s.io/legacy-cloud-providers/gce/gce.go

```diff
@@ -793,6 +793,29 @@ func (g *Cloud) HasClusterID() bool {
 	return true
 }
 
+// GetBasePath returns the compute API endpoint with the `projects/` element
+// compute API v0.36 changed basepath and dropped the `projects/` suffix, therefore suffix
+// must be added back when generating compute resource urls.
+func (g *Cloud) GetBasePath() string {
+	basePath := g.ComputeServices().GA.BasePath
+	return GetProjectBasePath(basePath)
+}
+
+// GetProjectBasePath returns the compute API endpoint with the `projects/` element.
+// The suffix must be added when generating compute resource urls.
+func GetProjectBasePath(basePath string) string {
+	if !strings.HasSuffix(basePath, "/") {
+		basePath += "/"
+	}
+	// Trim  the trailing /, so that split will not consider the last element as empty
+	elements := strings.Split(strings.TrimSuffix(basePath, "/"), "/")
+
+	if elements[len(elements)-1] != "projects" {
+		return fmt.Sprintf("%sprojects/", basePath)
+	}
```

Comment:
splitting just to check the last element is expensive. since we've ensured the path ends in `/` already, can simplify this to:

```go
if !strings.HasSuffix(basePath, "/projects/") {
   basePath += "projects/"
}
```